eclass/java-utils-2.eclass

```diff
@@ -1972,7 +1972,10 @@ etestng() {
 # src_prepare Searches for bundled jars
 # Don't call directly, but via java-pkg-2_src_prepare!
 java-utils-2_src_prepare() {
-	eapply_user
+	case ${EAPI} in
+		[678]) eapply_user ;;
```

Comment:
```suggestion
		[678]) if [[ $JAVA_UTILS_DEFAULT_SRC_PREPARE ]]; then default_src_prepare; else eapply_user; fi ;;
```

Reply:
What is the purpuse of `[[ $JAVA_UTILS_DEFAULT_SRC_PREPARE ]]`, when would it apply?
Guess I need an example to understand, and maybe also some words to add to the commit message.

Reply:
If  `JAVA_UTILS_DEFAULT_SRC_PREPARE` is set, ebuilds can opt-in to the new behavior. Adding this eclass variable requires a documentation block:

```
# @ECLASS_VARIABLE: JAVA_UTILS_DEFAULT_SRC_PREPARE
# @DEFAULT_UNSET
# @DESCRIPTION:
# If set, then the elcass will invoke default_src_prepare as part
# of its src_prepare function. This may be only set if the ebuild's
# EAPI is 6-8. For all other supported EAPI's of this class, the
# behavior that is enabled by setting this variable is the default
# behavior.
#
# Example value:
# @CODE
# JAVA_UTILS_DEFAULT_SRC_PREPARE=1
# @CODE
```

Reply:
@Flowdalic thanks for explaining. actually I don't see any benefit in that proposal, and it would even need an avoidable additional eclass variable.

Reply:
JAVA_UTILS_DEFAULT_SRC_PREPARE is based on the assumption that EAPI 9 will not appear soon. I am not sure, however, about the actual state of EAPI 9, but I wouldn't be surprised if we don't see EAPI 9 in 2024. 

That said, introducing JAVA_UTILS_DEFAULT_SRC_PREPARE was merely a suggestion to point out how ebuilds could benefit from the change without breaking the eclass' API or waiting for EAPI 9 to arrive. I don't have strong feelings about it, although it seems sensible to me, given I understand the situation correctly.

Reply:
Thanks again.  It is good to know about alternative options which all have their advantages and disadvantages.  At the end it's almost regardless which option we chose.  As another option I also [prepared a script months ago](https://github.com/gentoo/gentoo/pull/20608#issuecomment-1423022113), advantage: `no need to wait for EAPI 9`, disadvantage: `needs to find someone to review, test and merge that mass change`.

I still prefer to have it as was before [the reversal](https://github.com/gentoo/gentoo/files/10688454/prepare_default.txt), of course with `s/${EAPI:-0}/${EAPI}/`.
Would we need to send to ML?

Reply:
> disadvantage: `needs to find someone to review, test and merge that mass change`.

Another disadvantage of breaking the eclass API and doing a mass change on the ebuilds is that the change only affects ::gentoo. Overlays are not affected, and a while ago we had some rather big overlays with many java ebuilds. Although Gentoo often operates as ::gentoo would be the only thing in existence, I would like to avoid imposing a burden on overlays when possible.



> I still prefer to have it as was before [the reversal](https://github.com/gentoo/gentoo/files/10688454/prepare_default.txt),

Sure, we can merge this PR as is without ML review as it just restores the previous state.